PowerUp/app/src/main/java/powerup/systers/com/utils/DbResourceUtils.java

```diff
@@ -0,0 +1,132 @@
+package powerup.systers.com.utils;
+
+import android.content.Context;
+
+import powerup.systers.com.R;
+import powerup.systers.com.data.DataSource;
+import powerup.systers.com.data.IDataSource;
+
+public class DbResourceUtils {
+
+    private DataSource source;
+    private Context context;
+    public DbResourceUtils(DataSource source, Context context) {
+        this.source = source;
+        this.context = context;
+    }
+
+    public int getEyeResourceId() {
+        final int[] eye = new int[1];
+        String eyeImageName = context.getResources().getString(R.string.eye);
+        source.getAvatarEye(new IDataSource.LoadIntegerCallback() {
+            @Override
+            public void onResultLoaded(int value) {
+                eye[0] = value;
+            }
+        });
+        eyeImageName = eyeImageName + eye[0];
+        R.drawable ourRID = new R.drawable();
+        java.lang.reflect.Field photoNameField;
+        try {
+            photoNameField = ourRID.getClass().getField(eyeImageName);
+            //returning eye resource id
+            return photoNameField.getInt(ourRID);
+        } catch (NoSuchFieldException | IllegalAccessException
+                | IllegalArgumentException error) {
+            error.printStackTrace();
```

Comment:
This, won't give any info to debugger, if the application fails here, better to separate each exception, if needs to be logged with appropriate message, as if why exception came.
Ideally, utils should throw exception to caller and let them handle the way its appropriate, either logging or catching or throwing back to caller again.